src/reactA11yMouseEventHasKeyEventRule.ts

```diff
@@ -0,0 +1,109 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getJsxAttributesFromJsxElement } from './utils/JsxAttribute';
+
+const MOUSE_EVENTS: {
+    onMouseOver: {
+        value: 'onmouseover';
+        jsxValue: 'onMouseOver';
+    };
+    onMouseOut: {
+        value: 'onmouseout';
+        jsxValue: 'onMouseOut';
+    };
+} = {
```

Comment:
Seems like these literals don't get inferred by the compiler. Wondering if there's a less verbose way to declare these.

Reply:
How about creating an interface for the two objects containing `value` and `jsxValue`?

Alternately, perhaps with the new `as const` (`const` assertions)? https://devblogs.microsoft.com/typescript/announcing-typescript-3-4-rc/ Great timing if so!

Suggestion: upgrade tslint-microsoft-contrib to the 3.4 RC, and if that solves your use case, we can put tslint-microsoft-contrib onto that version before this PR goes in? /cc @IllusionMH

Reply:
Interface will have type `string` instead of literals.

Until 3.4 arrives explicit type assertions to the same literal might be a better way
```ts
const MOUSE_EVENTS = {
  onMouseOver: {
    value: 'onmouseover' as 'onmouseover',
    jsxValue: 'onMouseOver' as 'onMouseOver'
  },
  onMouseOut: {
    value: 'onmouseout' as 'onmouseout',
    jsxValue: 'onMouseOut' as 'onMouseOut'
  }
};
```

Frankly speaking, I don't know if it safe to jump 3 versions ahead of `tslint` package with shared dependency on `tsutils` and yet unknown Prettier and TSLint support for mentioned `const` type assertions.
If there shouldn't be any problems (shouldn't be a big problem with manual AST traversing) - I have no objections to upgrade TS.

Reply:
This complains about `as-cast` 🤔

Reply:
Classic TSLint...